### PR TITLE
tests: Add missing subpass pipeline barrier tests

### PR DESCRIPTION
**Pull request by spencer-lunarg**
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/5631

Confirmed the VUs are actually done and wrote extra tests to confirm

### CODE REVIEW

File: tests/unit/subpass.cpp [hunk 1]
```diff
@@ -1261,4 +1261,47 @@ TEST_F(NegativeSubpass, FramebufferNoAttachmentsSampleCounts) {
     m_errorMonitor->SetDesiredError("VUID-VkGraphicsPipelineCreateInfo-subpass-00758");
     pipe.CreateGraphicsPipeline();
     m_errorMonitor->VerifyFound();
-}
\ No newline at end of file
+}
+
+// Need clarification about VUID-vkCmdPipelineBarrier2-None-07893
+TEST_F(NegativeSubpass, DISABLED_DependencyViewLocalSubpass) {
```
Comment (spencer-lunarg) [inline tests/unit/subpass.cpp:1267]: I will follow-up with this next week, might need slight tweaking of the VU

File: tests/unit/dynamic_rendering_local_read.cpp [hunk 2]
```diff
@@ -1106,3 +1078,82 @@ TEST_F(NegativeDynamicRenderingLocalRead, AttachmentLocationsMax) {
...
+    memory_barrier_2.dstAccessMask = VK_ACCESS_2_COLOR_ATTACHMENT_READ_BIT;
+
+    VkDependencyInfoKHR dependency_info = vku::InitStructHelper();
+    dependency_info.dependencyFlags = VK_DEPENDENCY_VIEW_LOCAL_BIT;
+    dependency_info.memoryBarrierCount = 1;
+    dependency_info.pMemoryBarriers = &memory_barrier_2;
+    dependency_info.bufferMemoryBarrierCount = 0;
+    dependency_info.imageMemoryBarrierCount = 0;
+
+    m_errorMonitor->SetDesiredError("VUID-vkCmdPipelineBarrier2-dependencyFlags-07891");
```
Comment (artem-lunarg) [inline tests/unit/dynamic_rendering_local_read.cpp:1104]: Is it something from the spec that we need to test this VU with local read?

**Review by artem-lunarg — APPROVED** [review]

File: tests/unit/dynamic_rendering_local_read.cpp (hunk 2 above)
Comment (spencer-lunarg) [reply tests/unit/dynamic_rendering_local_read.cpp:1104]: yes, so you will hit `VUID-vkCmdPipelineBarrier2-None-09553` if you try to do a pipeline barrier in a renderpass

actually the `DependencyViewLocalOutsideRendering` **could** be in another file, but it made sense to keep these together

Comment (Williams23678) [reply tests/unit/dynamic_rendering_local_read.cpp:1104]: How can you make 10k in a day?

**Pull request merged**